tests/providers/azure/azure_test.py

```diff
@@ -13,3 +13,410 @@
+
+    # If we specify a resource group name, the 'list' method should be called.
+    FAKE_ACCOUNT.ListInstances(
+        resource_group_name=instance.resource_group_name)
+    mock_request.assert_called_with(
+        mock.ANY, 'list', {'resource_group_name': instance.resource_group_name})
+
+  @mock.patch('libcloudforensics.providers.azure.internal.common.ExecuteRequest')
+  @typing.no_type_check
+  def testListDisks(self, mock_request):
+    """Test that disks of an account are correctly listed."""
+    mock_request.return_value = MOCK_REQUEST_DISKS
+
+    # If we don't specify a resource group name, the 'list' method should be
+    # called.
+    disks = FAKE_ACCOUNT.ListDisks()
+    mock_request.assert_called_with(mock.ANY, 'list')
+    self.assertEqual(2, len(disks))
+    self.assertIn('fake-disk-name', disks)
+    disk = disks['fake-disk-name']
+    self.assertEqual('fake-disk-name', disk.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-disk-name', disk.resource_id)
+    self.assertEqual('fake-resource-group', disk.resource_group_name)
+    self.assertEqual('fake-region', disk.region)
+    self.assertEqual(['fake-zone'], disk.zones)
+
+    # If we specify a resource group name, the 'list_by_resource_group' method
+    # should be called.
+    FAKE_ACCOUNT.ListDisks(resource_group_name=disk.resource_group_name)
+    mock_request.assert_called_with(
+        mock.ANY,
+        'list_by_resource_group',
+        {'resource_group_name': disk.resource_group_name})
+
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.ListInstances')
+  @typing.no_type_check
+  def testGetInstance(self, mock_list_instances):
+    """Test that a particular instance from an account is retrieved."""
+    mock_list_instances.return_value = MOCK_LIST_INSTANCES
+    instance = FAKE_ACCOUNT.GetInstance('fake-vm-name')
+    self.assertEqual('fake-vm-name', instance.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-vm-name', instance.resource_id)
+    self.assertEqual('fake-resource-group', instance.resource_group_name)
+    self.assertEqual('fake-region', instance.region)
+    self.assertEqual(['fake-zone'], instance.zones)
+
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.ListDisks')
+  @typing.no_type_check
+  def testGetDisk(self, mock_list_disks):
+    """Test that a particular disk from an account is retrieved."""
+    mock_list_disks.return_value = MOCK_LIST_DISKS
+    disk = FAKE_ACCOUNT.GetDisk('fake-disk-name')
+    self.assertEqual('fake-disk-name', disk.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-disk-name', disk.resource_id)
+    self.assertEqual('fake-resource-group', disk.resource_group_name)
+    self.assertEqual('fake-region', disk.region)
+    self.assertEqual(['fake-zone'], disk.zones)
+
+  @mock.patch('azure.mgmt.compute.v2020_05_01.operations._disks_operations.DisksOperations.create_or_update')
```

Comment:
Oof, this sounds terrible. Are we going to have to change this mock whenever they update the library?

Reply:
Welcome to the delicious world of Azure's Python SDK :P 
Maybe there's some way to get whatever version is being used in code directly and then format the `mock.patch` statement accordingly? This would also allow us to un-pin the requirement. Either way, it'll be messy.

Reply:
Nah it's fine. But is this module intended to be accessed by library users? Otherwise it might be better to mock a higher-level function

Reply:
I think so, since this is the actual function that the documentation asks you to call